pkg/build/controller/factory/factory.go

```diff
@@ -305,11 +313,20 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 		),
 		Handle: func(obj interface{}) error {
 			imageRepo := obj.(*imageapi.ImageStream)
-			return imageChangeController.HandleImageRepo(imageRepo)
+			return imageChangeController.HandleImageStream(imageRepo)
 		},
 	}
 }
 
+func (factory *ImageChangeControllerFactory) waitForSyncedStores() {
+	for !factory.BuildConfigIndexSynced() {
```

Comment:
yeah i'm not signing up for rewriting the controllers just yet, that's a much larger piece of work so for now i just want to fix the performance problem.

when i remove the sync check, it actually does work, so the cache is getting properly populated.  I couldn't tell from your other PR what is supposed to trigger the cache to be initialized..I assumed since it was a shared informer, that should already be happening even if i'm not using it...